### PR TITLE
Issue 602: Add diagram to display SU usage by user on Allocation Detail page

### PR DESCRIPTION
**Pull request by haudquaquam**
**Changes**
- Adds a pie chart that displays the SU usage broken down by user. Hovering over a slice brings up the username, used SUs, and percentage of total used for users with positive SU usage.
- Added function in `allocation/utils.py` to generate the pie chart data.
- New `context_data` attribute for `pie_data` in `allocation/views.py`.
- New `drawPies` function in `allocation_detail.html` draws the pie chart, which is inserted right under the existing "Service Units" gauge graph.
- Legend appears to the right of the pie chart in columns.
- When there are no users with non-zero usage, displays a message indicating there is no data available.

**Testing**
- Manual testing can be done by navigating to any project's Allocation Detail page and ensuring things look okay. Good examples (BRC) are:
  - Project with many active users: `allocation/450/`
  - Project with average amount of users: `allocation/369/`
  - Project with one active user: `allocation/11/`
  - Project with no active users: `allocation/2/`

### CODE REVIEW

File: coldfront/core/allocation/templates/allocation/allocation_detail.html [hunk 1]
```diff
@@ -348,7 +361,26 @@ <h2 class="d-inline"><i class="fas fa-users" aria-hidden="true"></i> Messages fr
...
+  function drawPies(pie_data) {
+    if (pie_data.columns[0].length > 1) {
+      var chart = c3.generate({
+        bindto: '#user-pie',
+        pie: {
+          title: "Service Usage by User"
+        },
+        data: pie_data,
+        legend: {
+            position: 'right',
```
Comment (matthew-li) [inline coldfront/core/allocation/templates/allocation/allocation_detail.html:376]: Too much indentation?

**Review by matthew-li — CHANGES_REQUESTED** [review]
This looks great, thanks!

We don't need to sort by descending usage, given that hovering over the slice highlights the user in the legend.

Can you confirm: Are the users who have been removed from the project (see the table below "Users in Allocation") but have non-zero usage reflected in the chart? If not, they should be. (I think they already are, but haven't looked for an example.)

**Comment by matthew-li** [comment]
I've also tested that it works on an LRC box.

**Comment by haudquaquam** [comment]
Just confirmed that a removed user's SU usage still shows up on the graph. Addressed the other comments too, thanks Matthew!

**Pull request merged**